tests/integration/test_workflow_db_mgr.py

```diff
@@ -127,3 +128,53 @@ async def test_workflow_param_rapid_toggle(
 
         w_params = dict(schd.workflow_db_mgr.pri_dao.select_workflow_params())
         assert w_params['is_paused'] == '0'
+
+
+async def test_record_only_non_clock_triggers(flow, run, scheduler):
+    """Database does not record wall_clock xtriggers.
+
+    https://github.com/cylc/cylc-flow/issues/5911
+
+    Includes:
+        - Not in DB: A normal wall clock xtrigger (wall_clock).
+        - In DB: An xrandom mis-labelled as wall_clock trigger  DB).
+        - Not in DB: An execution retry xtrigger.
+
+    @TODO: Refactor to use simulation mode to speedup after Simulation
+    mode upgrade bugfixes: This should speed this test up considerably.
+    """
+    id_ = flow({
+        "scheduler": {
+            'cycle point format': '%Y'
+        },
+        "scheduling": {
+            "initial cycle point": "1348",
+            "xtriggers": {
+                "another": "xrandom(100)",
+                "wall_clock": "xrandom(100, _=Not a real wall clock trigger)",
+                "real_wall_clock": "wall_clock()"
+            },
+            "graph": {
+                "R1": "@another & @wall_clock & @real_wall_clock => foo"
+            }
+        },
+        'runtime': {
+            'foo': {
+                'execution retry delays': 'PT0S',
+                'script': (
+                    'test $CYLC_TASK_SUBMIT_NUMBER == 1 && exit 1 || exit 0')
```

Comment:
This will be do-able in simulation mode after the simulation mode refactoring, which should speed this test up rather a lot.

Reply:
Shell interaction is the line in the sand between integration and functional tests.

But I expect it can be dispensed with, here's one idea, get the simulated task to wait so you can feed the failed message into the system manually:

```ini
[runtime]
  [[foo]]
     [[[simulation]]]
       default run length = PT1M
```

```python
# fail a simulated task
schd.task_events_mgr.process_message(itask, 'error', 'failed')
```

Reply:
@oliver-sanders  - Using simulation mode isn't possible because of the bug outlined in https://github.com/cylc/cylc-flow/issues/5935.